### ISSUE TITLE
bracken-build "database library does not exist" for custom database

### ISSUE THREAD

**Issue by adec37** [jenniferlu717/Bracken#199]
I used the GTDB database (https://gtdb.ecogenomic.org/)  for kraken2 which consist of 'hash.k2d' 'opts.k2d' 'seqid2taxid.map' and 'taxo.k2d' files. I am now trying to run bracken-build as follows: 

`/users/a/d/adecola/anaconda3/envs/bracken/bracken-build -d /users/a/d/adecola/anaconda3/envs/kraken2/GTDB/ -k 35 -l 2000 -x /users/a/d/adecola/anaconda3/bin/
`
I am getting the following output: 

`Selected Options:
       kmer length = 35
       read length = 2000
       database    = /users/a/d/adecola/anaconda3/envs/kraken2/GTDB/
       threads     = 1
 Checking for Valid Options...
ERROR: Database library /users/a/d/adecola/anaconda3/envs/kraken2/GTDB/library does not exist`

I do not have the 'library' folder that is generated by the kraken2 standard database as I downloaded an external database. Any advice is much appreciated.

**Comment by apredeus** [comment]
Yep, same error here - downloaded `Kraken2` database from the official page. There are pre-calculated `Bracken` files in there, but no "library" folder. I don't think I've ever used a Kraken2 DB with a library folder?

**Comment by jenniferlu717** [comment]
You cannot make the database files without the library/ folder, so whoever built the original Kraken database will need to make the pre-calculated Bracken files.

**Comment by nityendra21** [comment]
Facing the exact issue as @jtsteyer93 except I have the Standard-8 db.

**Comment by merytouceda** [comment]
I am having the same issue as @jtsteyer93, has anyone been able to solve it?

**Comment by JuliaMcGonigle** [comment]
It's solved. To build the bracken database ref file with bracken-build, you need the library/ and taxonomy/ folders from the kraken-build commands. The taxonomy is easy to reproduce, however the library/ folder is I think a bunch of kmer hash files that are specific to your database and cannot be reproduced without the reference genomes used to build the kraken database (i.e. those  'hash.k2d' 'opts.k2d' 'seqid2taxid.map' and 'taxo.k2d' files). See lines 174-186 in bracken-build script (https://github.com/jenniferlu717/Bracken/blob/master/bracken-build).

The workflow is described in the Bracken documentation: build the kraken database, build the ref files for bracken, then classify with kraken, run bracken. Warning: do not run kraken-build --clean before building all the required bracken files as this will remove the contents of /library.

**Comment by JuliaMcGonigle** [comment]
Yeah I would agree that the bracken build files should have just been made and released with the kraken database k2d files. I don't think there is a way to get them from just the k2d files after digging around in the bracken-build script. Seems odd they weren't released together when these programs rely on each other and improve results. Re-building is sadly the only solution to this issue.

I was just throwing it out as a solution for folks who seem to still be having this problem if they still want to use bracken for their analysis, as I just recently also came across it. I used rsync to get my genomes from NCBI RefSeq/GenBank rather than the standard db download through kraken-build and that worked ok for me with minor issues.

**Comment by lauramason326** [comment]
If I download the library to run bracken build, will bracken include my custom files as well? Asking because the default fungal DB has way fewer genomes than my custom one, and I want to make sure they are not missed.
